pkg/cmd/server/kubernetes/network/network.go

```diff
@@ -232,17 +236,30 @@ func (c *NetworkConfig) RunProxy() {
 }
 
 // getNodeIP is copied from the upstream proxy config to retrieve the IP of a node.
-func getNodeIP(client kv1core.CoreV1Interface, hostname string) net.IP {
-	var nodeIP net.IP
-	node, err := client.Nodes().Get(hostname, metav1.GetOptions{})
-	if err != nil {
-		glog.Warningf("Failed to retrieve node info: %v", err)
-		return nil
+func getNodeIP(client kv1core.CoreV1Interface, hostname string) (net.IP, error) {
+	var node *v1.Node
+	var nodeErr error
+
+	// We may beat the thread that causes the node object to be created,
+	// so if we can't get it, then we need to wait.
+	// This will wait 2, 4, 8, ... 256 seconds, for a total of 8.5 mins
```

Comment:
For the first iteration it will not wait, so the times will be
0, 2, 4, 8, ... 128 secs, for a total of approx 4 mins

I found this issue in other places in SDN code and I plan to submit a fix in a separate PR.

Reply:
Ah, thanks!  I'll update.